app/helpers/hearing_paginator.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+# Creates a chronologically sorted set of "items"
+# with next and previous navigation helpers.
+#
+# Next and previous correlate to navigation over all
+# a cases' hearing's hearing days.
+#
+class HearingPaginator
+  include ActionView::Helpers
+  include Rails.application.routes.url_helpers
+
+  PageItem = Struct.new(:id, :hearing_date)
+
+  # rubocop:disable Rails/HelperInstanceVariable
+  def initialize(prosecution_case, options = {})
+    @prosecution_case = prosecution_case
+    @current_page = options.fetch(:page, nil).to_i
+  end
+  # rubocop:enable Rails/HelperInstanceVariable
+
+  def items
+    hearing_items_by_datetime
+  end
+
+  def current_item
+    items[current_page || first_page]
+  end
+
+  attr_writer :current_page
```

Comment:
> Move attr_writer to the top of the class.

I kept this together with the attribute reader version for clarity. In terms of it being at the top what is the advantage? and what would the top be? above the `include` statements and Struct definition, which does not seem right to me, or below.

> Also, if it is not required for the public API of the class consider removing it and using @current_page instead.

true, albeit that the public API is somewhat in flux at the moment (there are other methods which do not need to be public too). current_page is used in the tests for convenience at the moment so many would have to be rewritten too - I will look at cleaning this up.

Reply:
have removed the `current_page` `attr_writer` now and rewritten the tests.

Reply:
> In terms of it being at the top what is the advantage?

Only because I thought that it is generally good practice to have them listed at the top of a class. (Did I imagine that?)

> > Also, if it is not required for the public API of the class consider removing it and using @current_page instead.
> 
> true, albeit that the public API is somewhat in flux at the moment

Having things public when they don't need to be could make it trickier to change later.

> current_page is used in the tests for convenience at the moment so many would have to be rewritten too

I did see that and I think it would be better to have the class created as it would actually happen (by passing the correct parameters) rather than modifying internal values. If you modify internal values then there could come a time where we are testing situations that cannot occur due to changes in the interface. (I believe you have already updated the tests along these lines)

Reply:
> have removed the `current_page` `attr_writer` now and rewritten the tests.